### DISCUSSION TITLE
how to split corpus into train/dev ?

### DISCUSSION THREAD

**Discussion by d5555** [explosion/spaCy#7822] [New Features & Project Ideas]
Is it possible to add a feature to `Corpus` to get iterable according persentage from the start and from the end  so I can devide my dataset into train and dev ?
`train=Corpus (DATASET_Path ,   0.8 )` # 80%  from the start 
`dev=Corpus (DATASET_Path ,   -0.2 )` # 20%  from the end

There is an option `limit` but it won't work if we need to set a limit from the end (to get `dev` Corpus)

## Your Environment
- Operating System:
- spaCy Version Used: 3.0.5

**Comment by adrianeboyd** [comment]
We can consider whether more built-in options make sense here.

Currently your best option is write your own corpus reader. Here's what the provided one looks like:

https://github.com/explosion/spaCy/blob/ed561cf428494c2b7a6790cd4b91b5326102b59d/spacy/training/corpus.py#L22-L39

You'd want to modify the options and how the corpus is returned to split a single file into partitions.

**Comment by mbrunecky** [comment]
Doing the first 80% last 20% split assumes the corpus is already thoroughly randomized. But than it should be easy enough to just split the DATASET_path into two directories by hand (assuming there are multiple smaller docBins).
Another approach to splitting the dataset into 'train' and 'dev' is getting a random value for each doc and  if the random value is >= 0.8 the doc goes into 'train' else into 'dev'.
I am not sure this enhancements is needed, because (at least in my experience) preparing the dataset takes a lot of effort (I am creating docBins with my code), and splitting it into train/dev is just a small piece of that.

But for those using the convert utility perhaps this could be a convert utility feature...
